lab/lab-1.md

```diff
@@ -0,0 +1,209 @@
+
+- `app=backend`
+- `version=v1`
+
+Put the deployment into the newly created namespace `demoapp`:
+
+```sh
+[centos@node-0 example-app-backend]$ kubectl apply -f deployment_v1.yaml
+```
+
+If you need more information about how to create a kubernetes deployment manifest, check the docs [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#creating-a-deployment). The deployment should run the **backend** docker image in version `v1.0.0`.
+
+## Expose the backend internally
+
+As we want to access the backend pods from our frontend we have to expose it internally.
+Which is Kubernetes object is used to expose a deployment on the cluster?
+And how can we make it only internally available? Is there a special type for the resource we are looking for?
+Use both labels(`app`, `version`) for the pod selection.
+
+## Build frontend
+
+Navigate to your user home directory or whereever you want to store the source code of the frontend serivce.
+Now checkout the sample repository on your `node-0` via:
+
+```sh
+[centos@node-0 ~]$ cd
+[centos@node-0 ~]$ git clone --branch v1.0.0 https://github.com/hulk0301/example-app-frontend.git
+[centos@node-0 ~]$ cd example-app-frontend
+[centos@node-0 example-app-frontend]$
+```
+
+Then build the docker image using the provided Dockerfile.
+Use `frontend:v1.0.0` as tag for your image and push it to DockerHub(you should have already an account, if not created one).
+
+## Deploy the frontend to Kubernetes
+
+Create a deployment manifest(`deployment_v1.yaml`) and apply it on the Kubernetes cluster using `kubectl`. Don't forget to specify some meaningful labels:
+
+- `app=frontend`
+- `version=v1`
+
+Put the deployment also in the namespace `demoapp`(The namespace was create in the `Deploy the backend to Kubernetes` section):
+
+```sh
+[centos@node-0 example-app-frontend]$ kubectl apply -f deployment_v1.yaml
+```
+
+If you need more information about how to create a kubernetes deployment manifest, check the docs [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#creating-a-deployment). The deployment should run the **frontend** docker image in version `v1.0.0`.
+
+Don't forget to set the environment variable `API_URL` to a valid URL(`http://<MY_BACKEND_HOST>:<MY_BACKEND_PORT>`).
+How can you configure the environment for your deployment? Check out the docs about environment for containers [here](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/#define-an-environment-variable-for-a-container).
+
+## Expose the frontend service to the public internet
+
+Now we want to expose our frontend to world, so that everyone can use our application. Therefore we have to make the frontend publicly available. How can we do this? What options do we have and what is the easiest way to do this?
```

Comment:
```suggestion
Now we want to expose our frontend to the world, so that everyone can use our application. Therefore we have to make the frontend publicly available. How can we do this? What options do we have and what is the easiest way to do this?
```